drivers/tapdisk-nbdserver.c

```diff
@@ -429,21 +430,63 @@ receive_newstyle_options(td_nbdserver_t *server, int new_fd, bool no_zeroes)
 			ERR("NBD_OPT_INFO: not implemented");
 			break;
 		case NBD_OPT_GO:
-		{   
+		{
 			uint16_t flags;
-			INFO("Processing NBD_OPT_EXPORT_NAME");
+			uint32_t exportnamelen;
+			uint16_t nrInfoReq;
+
+			INFO("Processing NBD_OPT_GO");
+
+			if (recv(new_fd, &exportnamelen, sizeof(exportnamelen), 0) == -1) {
+				ret = -1;
+				goto done;
+			}
+			exportnamelen = be32toh(exportnamelen);
+
+			exportname = malloc(exportnamelen + 1);
```

Comment:
While the other side is presumably trusted, it is probably worth sanity checking exportnamelen just in case.

Reply:
Only moderately trusted so yes it does need input validation.